### PR TITLE
gnrc_netif: Use fmt for bytes to hex string

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Small code dedup by using fmt  instead of a custom function for printing a hex string

### Issues/PRs references

None

### CODE REVIEW

File: sys/net/gnrc/netif/gnrc_netif.c [hunk 1]
```diff
@@ -347,20 +348,14 @@ gnrc_netif_t *gnrc_netif_get_by_pid(kernel_pid_t pid)
...
-    return (half_byte < 10) ? ('0' + half_byte) : ('a' + (half_byte - 10));
-}
-
 char *gnrc_netif_addr_to_str(const uint8_t *addr, size_t addr_len, char *out)
 {
     char *res = out;
 
     assert((out != NULL) && ((addr != NULL) || (addr_len == 0U)));
     out[0] = '\0';
     for (size_t i = 0; i < addr_len; i++) {
```
Comment (kaspar030) [inline sys/net/gnrc/netif/gnrc_netif.c]: why not just ```out += fmt_bytes_hex(out, addr, addr_len); *out = '\0';```?

**Comment by miri64** [comment]
> I can't really say that I have an opinion about this? Do you mind that it is changed to upper case here?

No, I'm just saying that we should check all our `pexpect` scripts if they still work with this PR and if they don't, adapt them accordingly.

**Comment by bergzand** [comment]
> No, I'm just saying that we should check all our `pexpect` scripts if they still work with this PR and if they don't, adapt them accordingly.

@miri64 Adapted them.

I've also rebased this PR because the gnrc_netif test was failing otherwise (issue unrelated to this PR).

**Comment by miri64** [comment]
I think you forgot to push ;-)

**Comment by bergzand** [comment]
> I think you forgot to push ;-)

Ssssssht :)

**Comment by PeterKietzmann** [comment]
Tests succeed on native and samr21-xpro. @kaspar030 mind to remove your change request?

**Comment by PeterKietzmann** [comment]
BTW: I've tried to incorporate the new labels

**Comment by PeterKietzmann** [comment]
@bergzand please sqash!

**Comment by bergzand** [comment]
> @bergzand please sqash!

Sqashed!

**Pull request merged**